worker/src/main/java/org/apache/celeborn/service/deploy/worker/memory/MemoryManager.java

```diff
@@ -557,6 +580,47 @@ public static void reset() {
     _INSTANCE = null;
   }
 
+  private void resumeByPinnedMemory(ServingState servingState) {
+    switch (servingState) {
+      case PUSH_AND_REPLICATE_PAUSED:
+        logger.info(
+            "Serving State is PUSH_AND_REPLICATE_PAUSED, but resume by lower pinned memory {}",
+            getNettyPinnedDirectMemory());
+        resumeReplicate();
+        resumePush();
+      case PUSH_PAUSED:
+        logger.info(
+            "Serving State is PUSH_PAUSED, but resume by lower pinned memory {}",
+            getNettyPinnedDirectMemory());
+        resumePush();
+    }
+  }
+
+  private boolean canResumeByPinnedMemory() {
+    if (pinnedMemoryCheckEnabled
+        && System.currentTimeMillis() >= pinnedMemoryNextCheckTime
+        && getAllocatedMemory() / (double) (maxDirectMemory) < pinnedMemoryResumeRatio) {
+      pinnedMemoryNextCheckTime += pinnedMemoryCheckInterval;
```

Comment:
`pinnedMemoryNextCheckTime` compute seems incorrect, We can Use `System.currentTimeMillis()` as last checkTime, then we can use `System.currentTimeMillis()-lastCheckTime >= pinnedMemoryCheckInterval` to check whether need resume.